network/src/sync/master.rs

```diff
@@ -80,7 +80,27 @@ impl SyncMaster {
     }
 
     async fn block_locator_hashes(&mut self) -> Result<Vec<Digest>> {
-        match self.node.storage.get_block_locator_hashes().await {
+        let forks_of_interest = self.node.expect_sync().consensus.scan_forks().await?;
+        let blocks_of_interest: Vec<Digest> = forks_of_interest.into_iter().map(|(_canon, fork)| fork).collect();
+        let mut tips_of_blocks_of_interest: Vec<Digest> = Vec::with_capacity(blocks_of_interest.len());
```

Comment:
```suggestion
        let blocks_of_interest = forks_of_interest.into_iter().map(|(_canon, fork)| fork);
        let mut tips_of_blocks_of_interest: Vec<Digest> = Vec::new();
```

Reply:
iter needs to be mut to be called. why remove the capacity?

Reply:
not sure which `mut` you have in mind (as the one here wasn't removed), but `blocks_of_interest` can just be an iterator; the capacity is removed, because small forks are not included, and I think it's the most usual case we'll encounter.